relevanceai/clusterer/clusterer.py

```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
+        # self.model.vector_fields = vector_fields
+
+        # self._insert_centroid_documents()
+
+        if verbose:
+            print(
+                "Build your clustering app here: "
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subpartialfit_predict_update(
+        self,
+        dataset,
+        vector_fields: list,
+        filters: list = None,
+        cluster_ids: list = None,
+        verbose: bool = True,
+    ):
+        """
+
+        Retrieve all the documents.
+
+        Parameters
+        ------------
+
+        dataset: Dataset
+            The dataset to call fit predict update on
+        vector_fields: list
+            The list of vector fields
+        filters: list
+            The list of filters
+
+        Example
+        ----------
+
+        .. code-block::
+
+
+        """
+        # Get data
+
+        self._init_dataset(dataset)
+
+        filters = [] if filters is None else filters
+        cluster_ids = [] if cluster_ids is None else cluster_ids
+        # Loop through each unique cluster ID and run clustering
+        parent_field = self._get_cluster_field_name(self.parent_alias)
+
+        print("Getting unique cluster IDs...")
+        unique_clusters = self.unique_cluster_ids(alias=self.parent_alias)
+
+        for i, unique_cluster in enumerate(tqdm(unique_clusters)):
+            cluster_filters = filters.copy()
+            cluster_filters += [
+                {
+                    "field": parent_field,
+                    "filter_type": "category",
+                    "condition": "==",
+                    "condition_value": unique_cluster,
+                }
+            ]
+            self.partial_fit_predict_update(
+                dataset=self.dataset_id,
+                vector_fields=vector_fields,
+                filters=cluster_filters,
+                verbose=False,
+            )
+
+        if verbose:
+            print(
+                "Build your clustering app here: "
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subfit_predict_documents(
+        self,
+        vector_fields,
+        filters: list,
+        cluster_ids: list = None,
+        verbose: bool = True,
+    ):
+        filters = [] if filters is None else filters
+        cluster_ids = [] if cluster_ids is None else cluster_ids
+        # Loop through each unique cluster ID and run clustering
+        parent_field = self._get_cluster_field_name(self.parent_alias)
+
+        print("Getting unique cluster IDs...")
+        if not cluster_ids:
+            unique_clusters = self.unique_cluster_ids(alias=self.parent_alias)
+        else:
+            unique_clusters = cluster_ids
+
+        for i, unique_cluster in enumerate(tqdm(unique_clusters)):
+            cluster_filters = filters.copy()
+            cluster_filters += [
```

Comment:
Nit: I think append is more idiomatic. Also, is there a need to make a copy of `filters`?

Reply:
Yep. Here is a code snippet to try out. 

```
a = ['a', 'b']
b = a
b += ['c']
a
```

Reply:
Ok, I see your point. I didn't think of the case in which a user would reuse a filter used in the argument.